coderd/templateversions.go

```diff
@@ -834,19 +837,79 @@ func (api *API) postTemplateVersionsByOrganization(rw http.ResponseWriter, r *ht
 	// Ensures the "owner" is properly applied.
 	tags := provisionerdserver.MutateTags(apiKey.UserID, req.ProvisionerTags)
 
-	file, err := api.Database.GetFileByID(ctx, req.FileID)
-	if errors.Is(err, sql.ErrNoRows) {
-		httpapi.Write(ctx, rw, http.StatusNotFound, codersdk.Response{
-			Message: "File not found.",
+	if req.ExampleID != "" && req.FileID != uuid.Nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "You cannot specify both an example_id and a file_id.",
 		})
 		return
 	}
-	if err != nil {
-		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
-			Message: "Internal error fetching file.",
-			Detail:  err.Error(),
+
+	var file database.File
+	var err error
+	// if example id is specified we need to copy the embedded tar into a new file in the database
+	if req.ExampleID != "" {
+		if !api.Authorize(r, rbac.ActionCreate, rbac.ResourceFile.WithOwner(apiKey.UserID.String())) {
+			httpapi.Forbidden(rw)
+			return
+		}
+		// ensure we can read the file that either already exists or will be created
+		if !api.Authorize(r, rbac.ActionRead, rbac.ResourceFile.WithOwner(apiKey.UserID.String())) {
+			httpapi.Forbidden(rw)
+			return
+		}
+
+		// lookup template tar from embedded examples
+		tar, err := examples.Archive(req.ExampleID)
+		if err != nil {
+			if xerrors.Is(err, examples.ErrNotFound) {
+				httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
```

Comment:
```suggestion
				httpapi.Write(ctx, rw, http.StatusNotFound, codersdk.Response{
```

Reply:
So I originally had this, but my thought was that a bad input into a form should probably be a 400, right?

Reply:
hmmmm that is a very philosophical point, I could go either way honestly